modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/DistributionZoneRebalanceEngine.java

```diff
@@ -334,6 +342,25 @@ public void onError(Throwable e) {
         };
     }
 
+    private int getCatalogVersionForCounter(int zoneId, int partId, long revision) {
+        try {
+            return metaStorageManager.get(catalogVersionKey(zoneId, partId), revision)
+                    .thenApply(entry -> {
+                        assert entry.value() != null : "Failed to find catalog version for table counters.";
+
+                        return bytesToInt(entry.value());
+                    })
+                    .thenCompose(storedCatalogVersion ->
+                            catalogService.catalogReadyFuture(storedCatalogVersion).thenApply(unused -> storedCatalogVersion)
+                    ).get();
```

Comment:
We will call getCatalogVersionForCounter only from within rebalance scheduler thread, right?